ads/inabox/inabox-messaging-host.js

```diff
@@ -224,12 +224,20 @@ export class InaboxMessagingHost {
       return this.iframeMap_[sentinel];
     }
 
-    // Walk up on the window tree until find host's direct child window
-    while (source.parent !== this.win_ && source !== this.win_.top) {
-      source = source.parent;
+    // Walk up on the window tree and verify that there is no cross-domain
+    // frame in between the source of the postMessage and the host doc.
+    // If there is, the host doc will not be able to accurately measure
+    // the creative's positions in the page, so return null.
+    let tempWin = source.parent;
+    while (tempWin !== this.win_ && tempWin !== this.win_.top) {
```

Comment:
Let's limit the amount of iterations (say 10) as we've seen situations where somehow windows get into an infinite loop.  If we hit the limit, return null.